### ISSUE TITLE
Questions on distributed agents

### ISSUE THREAD

**Issue by ethanluoyc** [deepmind/acme#133]
Hi!

I have been working on a distributed version of the JAX IMPALA agent. I found the distributed TF D4PG to be really helpful for guiding my implementation. However, after making a nearly complete implementation, I have the following questions and hope the Acme maintainers can provide some tips about them.

1. Creating reproducible environments with different processes. In D4PG, I saw that the agent takes in an environment factory with signature `environment_factory(training: bool)`. I would love to create environments with different seeds in a deterministic way in the actors and evaluators, but I have not found an API for this. Normally in distributed training one would seed the process based on some initial seed and the rank of the worker. However, I couldn't find an API for getting the "rank" of a process. Would the maintainers be able to share some tips on how to implement this behaviour?
2. I found that I cannot get a graceful shutdown as the reverb process does not seem to respect shutdown signals from LaunchPad. I have no idea why this is failing and it would be really useful if there are some tips regarding this. I would also prefer that LaunchPad does not kill the process after the timeout period by killing the process since I may want to perform some cleanup or continue another experiment in the same process. 
3. About logging. It would be really useful if Acme can support an experiment tracking platform such as wandb, I am not suggesting that Acme should have built-in integration with such platforms, but it should make it easy for such integrations to be possible. I find challenging to incorporate a custom logger. There is currently no API for sending data to a single logger process, and this makes it hard to adopt a common pattern in parallel training such as in MPI where the rank-zero process will be responsible for logging the results for the experiment.

**Comment by wcarvalho** [comment]
Also interested in answers to this

**Comment by qstanczyk** [comment]
Regarding question #2 - we have made a number of changes to Launchpad and Reverb integration, so it should not hang anymore. Could you give it a try?

**Comment by nikolamomchev** [comment]
@ethanluoyc I've submitted separate requests for 1) and 3) (we are still working on them) and my understanding is that 2) is no longer an issue. Closing this bug.

Duplicate of #171 and #172
